File: WallBreakers/BasicMath/SelfDividingNumbers.java
class Solution {
    public List<Integer> selfDividingNumbers(int left, int right) {
      ArrayList<Integer> results = new ArrayList<>();
      for (int curNum = left; curNum <= right; curNum++) {
        if (isSelfDividingNumber(curNum)) {
          results.add(curNum);
        }
      }
      return results;
    }
  
    private boolean isSelfDividingNumber(int num) {
      int curNum = num, digit;
      while (num > 0) {
        digit = num % 10;
        num /= 10;
        if (digit == 0 || curNum % digit != 0) {
          return false;
        }
      }
      return true;
    }
}

File: WallBreakers/BasicMath/SelfDividingNumbers.py
class Solution:
    def selfDividingNumbers(self, left: int, right: int) -> List[int]:
      def self_dividing_number(num):
        curNum = num
        while num > 0:
          digit = num % 10
          num = num // 10
          if digit == 0 or curNum % digit != 0:
            return False
        return True
        # for digit in str(num):
        #   if digit == '0' or num % int(digit) != 0:
        #     return False
        # return True
      
      results = []
      for num in range(left, right + 1):
        if self_dividing_number(num):
          results.append(num)
      return results